pandas/tests/indexes/period/test_indexing.py

```diff
@@ -685,3 +685,35 @@ def test_period_index_indexer(self):
         tm.assert_frame_equal(df, df.loc[list(idx)])
         tm.assert_frame_equal(df.iloc[0:5], df.loc[idx[0:5]])
         tm.assert_frame_equal(df, df.loc[list(idx)])
+
+    def test_get_loc_not_use_engine(self):
+        # issue 29234
+        idx = PeriodIndex(
```

Comment:
can you separate this out into 2 functions, one for errors and one for working cases.